jetty-start/src/test/java/org/eclipse/jetty/start/CommandLineBuilderTest.java

```diff
@@ -23,43 +28,87 @@ public class CommandLineBuilderTest
+        CommandLineBuilder cmd = new CommandLineBuilder();
+        cmd.addArg("java");
+        cmd.addArg("-Djetty.home", "/opt/jetty");
+        assertThat(cmd.toCommandLine(), is("java -Djetty.home=/opt/jetty"));
     }
 
     @Test
     public void testSimpleHomeWithSpace()
     {
-        CommandLineBuilder cmd = new CommandLineBuilder("java");
-        cmd.addEqualsArg("-Djetty.home", "/opt/jetty 10/home");
-        assertThat(cmd.toQuotedString(), is("java '-Djetty.home=/opt/jetty 10/home'"));
+        CommandLineBuilder cmd = new CommandLineBuilder();
+        cmd.addArg("java");
+        cmd.addArg("-Djetty.home", "/opt/jetty 10/home");
+        assertThat(cmd.toCommandLine(), is("java -Djetty.home='/opt/jetty 10/home'"));
     }
 
     @Test
     public void testEscapedFormattingString()
     {
-        CommandLineBuilder cmd = new CommandLineBuilder("java");
-        cmd.addEqualsArg("-Djetty.home", "/opt/jetty");
-        cmd.addEqualsArg("jetty.requestlog.formatter", "%{client}a - %u %{dd/MMM/yyyy:HH:mm:ss ZZZ|GMT}t \"%r\" %s %O \"%{Referer}i\" \"%{User-Agent}i\"");
-        assertThat(cmd.toQuotedString(), is("java -Djetty.home=/opt/jetty 'jetty.requestlog.formatter=%{client}a - %u %{dd/MMM/yyyy:HH:mm:ss ZZZ|GMT}t \"%r\" %s %O \"%{Referer}i\" \"%{User-Agent}i\"'"));
+        CommandLineBuilder cmd = new CommandLineBuilder();
+        cmd.addArg("java");
+        cmd.addArg("-Djetty.home", "/opt/jetty");
+        cmd.addArg("jetty.requestlog.formatter", "%{client}a - %u %{dd/MMM/yyyy:HH:mm:ss ZZZ|GMT}t \"%r\" %s %O \"%{Referer}i\" \"%{User-Agent}i\"");
+        assertThat(cmd.toCommandLine(), is("java -Djetty.home=/opt/jetty jetty.requestlog.formatter='%{client}a - %u %{dd/MMM/yyyy:HH:mm:ss ZZZ|GMT}t \"%r\" %s %O \"%{Referer}i\" \"%{User-Agent}i\"'"));
     }
 
     @Test
     public void testEscapeUnicode()
     {
-        CommandLineBuilder cmd = new CommandLineBuilder("java");
-        cmd.addEqualsArg("-Djetty.home", "/opt/jetty");
-        cmd.addEqualsArg("monetary.symbol", "€");
-        assertThat(cmd.toQuotedString(), is("java -Djetty.home=/opt/jetty monetary.symbol=€"));
+        CommandLineBuilder cmd = new CommandLineBuilder();
+        cmd.addArg("java");
+        cmd.addArg("-Djetty.home", "/opt/jetty");
+        cmd.addArg("monetary.symbol", "€");
+        assertThat(cmd.toCommandLine(), is("java -Djetty.home=/opt/jetty monetary.symbol='€'"));
+    }
+
+    public static Stream<Arguments> shellQuoting()
+    {
+        return Stream.of(
+            Arguments.of(null, null),
+            Arguments.of("", "''"),
+            Arguments.of("Hello", "Hello"),
+            Arguments.of("Hell0", "Hell0"),
+            Arguments.of("Hello$World", "'Hello$World'"),
+            Arguments.of("Hello\\World", "'Hello\\World'"),
+            Arguments.of("Hello`World", "'Hello`World'"),
+            Arguments.of("'Hello World'", "\\''Hello World'\\'"),
```

Comment:
This is weird. I would have expected to begin and end with a single quote.

Reply:
Escaping `'` only works outside of quotes so it could be `''\\''Hello World'\\'''`, but the leading and trailing `''` are just useless noise. 

If we really cared we would quote something like `'Hello'` as `\'Hello\'` and never enter the quotes.  But I'm guessing that most times there are quotes, then there will be a character than needs quoting.